src/rocm_docs/core.py

```diff
@@ -200,45 +213,95 @@ def _set_all_article_info(
     Add article info headers with general settings to all HTML pages
     except those in app.config.article_pages
     """
-    all_pages = _get_all_pages(app.config.html_output_directory)
+    (html_pages, source_map) = _get_all_pages(app.config.html_output_directory)
 
-    for page in all_pages:
+    for page in html_pages:
         # skip pages with specific settings
         if page in specific_pages:
             continue
 
-        # default to linux os
-        font_awesome_os = '<i class="fa-brands fa-linux fa-2xl"></i>'
-        if "linux" not in app.config.all_article_info_os:
-            font_awesome_os = ""
+        font_awesome_os = ""
+        if "linux" in app.config.all_article_info_os:
+            font_awesome_os += '<i class="fa-brands fa-linux fa-2xl fa-fw"></i>'
         if "windows" in app.config.all_article_info_os:
-            font_awesome_os += '<i class="fa-brands fa-windows fa-2xl"></i>'
+            font_awesome_os += '<i class="fa-brands fa-windows fa-2xl fa-fw"></i>'
+
+        page_key = Path(page).stem
+        if page_key in source_map.keys():
+            modified_path = source_map[Path(page).stem]
+        else:
+            modified_path = page
 
-        modified_info = article_info.replace("<!--fontawesome-->", font_awesome_os)
+        modified_info = article_info.replace("<!--osicons-->", font_awesome_os)
         modified_info = modified_info.replace("AMD", app.config.all_article_info_author)
-        modified_info = modified_info.replace("2023", app.config.all_article_info_date)
-        modified_info = modified_info.replace("5 min read", app.config.all_article_info_read_time)
+        modified_info = modified_info.replace("2023", _get_time_last_modified(modified_path))
+        modified_info = modified_info.replace("5 min read", _estimate_read_time(page))
         
         _write_article_info(page, modified_info)
 
 
-def _get_all_pages(output_directory: str) -> List[str]:
-    all_pages = []
+def _get_all_pages(output_directory: str):
+    html_pages = list()
+    source_map = dict()
+
     for root, _, files in os.walk(output_directory):
         for file in files:
             if file.endswith(".html"):
-                all_pages.append(os.path.join(root, file))
-    return all_pages
+                html_pages.append(os.path.join(root, file))
+
+    for root, _, files in os.walk("."):
+        for file in files:
+            if file.endswith(".rst") or file.endswith(".md"):
+                file_key = Path(file).stem
+                source_map[file_key] = os.path.join(root, file)
+    
+    return (html_pages, source_map)
+
+
+def _get_time_last_modified(path: str) -> str:
+    last_time_modified = os.path.getmtime(path)
+    global_time = time.gmtime(last_time_modified)
+    date_info = time.strftime("%B %d, %Y", global_time)
+    return date_info
```

Comment:
This is the file modification time, which will always be the build time on RTD as it does as a clean checkout every time it builds, and IIRC git does not set the mtime. If we want to automate this it should be based on the information from git, and not the filesystem.

Reply:
Thanks for pointing this out. I also only realized this after. 

Fixed to use git log in [a84f63f](https://github.com/RadeonOpenCompute/rocm-docs-core/pull/139/commits/a84f63fcfe0e77d60591a27d2e882c3bd812e212)